docs/upgrade.md

```diff
@@ -4,3 +4,5 @@
 - In rollingDeploy each node is update one by one, and incase any of the node goes in pending/crashing state during update the operator halts the update and does not update the other nodes. This requires manual intervation.
 - Default updates and cluster creation is in parallel. 
 - Operator supports ```HPA autosaling/v1beta1``` Spec in the nodeSpec for druid nodes. In case HPA deployed, HPA controller maintains the replica count/state for the particular statefulset referenced.  Refer to ```examples.md``` for HPA configuration. Prefered to scale only brokers using HPA.
+- During upgrade if sts is set to ordered ready, the sts controller will not recover from crashloopback state. The issues is referenced [here](https://github.com/kubernetes/kubernetes/issues/67250), and here's a reference [doc](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#forced-rollback)
+- How operator solves this is using the ```forcedRollbackSts``` key, if this key is set to ```true``` the operator will delete the sts pod if its in crashloopback state. Example Scenario: During upgrade, user rolls out a faulty configuration causing the historical pod going in crashing state, user rolls out a valid configuration, the new configuration will not be applied unless user manual delete pods, so solve this scenario operator shall delete the pod automatically without user intervention. ```NOTE: User must be aware of this feature, there might be cases where crashloopback might be caused due probe failure, fault image etc, the operator shall keep on deleting on each re-concile loop. Default Behavior is False ```
```

Comment:
I think default behavior should be true.

Also, we are not doing any rolling back of the sts and the flag name says otherwise, we can probably call it `forceDeleteStsPodOnError`

Reply:
sure ill change the flag name

BTW i feel default behavior should be false, since this a specific scenario and requires user to be aware of this, thats why i added this in notes
```
User must be aware of this feature, their might be cases where crashloopback might be caused due probe failure, fault image etc, the operator shall keep on deleting on each re-concile loop
```

Reply:
I understand that but there is still the assumption that it can't have a negative effect or else why would anyone able enable it?

or do you think there could be adverse effects of deleting pods selected by our intricate set of checks in `checkCrashStatus` ?

Reply:
@himanshug have done the changes, keeping default to true.

>  do you think there could be adverse effects of deleting pods selected by our intricate set of checks in checkCrashStatus

the only thing here to be noted is, whenever your pod goes into ERROR, CRASHLOOP, IMAGEPULLBACK, operator will keep on deleting the pod. and thats what we want ideally. i m fine keeping it to true